Basic.Azure.Storage/Communications/Core/RequestBase.cs

```diff
@@ -147,30 +146,65 @@ private static void ApplyAuthorizationHeader(StorageServiceType serviceType, Web
         private async Task<Response<TPayload>> SendRequestWithRetryAsync(ConcurrentDictionary<string, string> responseCodeOverridesForApiBugs = null)
         {
             var numberOfAttempts = 0;
+            var retryStack = new Stack<Exception>();
```

Comment:
So now I collect a stack of exceptions and add the stack to the retried or other exception that is thrown below.